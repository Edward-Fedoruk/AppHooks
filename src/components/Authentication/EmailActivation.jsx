import React from 'react'
import withBackground from './withBackground'
import { withStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import PropTypes from 'prop-types'

const styles = ({ breakpoints }) => ({
  paper: {
    padding: "36px 50px",
    width: "500px",
    height: "fit-content",
    marginTop: "8%",

    [breakpoints.down(500)]: {
      width: "90%",
      padding: "36px 15px",
      alignSelf: "center",  
    }
  },

  title: {
    fontWeight: "500",
    fontSize: "20px",
    color: "#5A5B5F",
    marginBottom: "25px"
  },

  p: {
    fontWeight: "400",
    fontSize: "16px",
    color: "#5A5B5F",
    lineHeight: "23px",
    marginTop: "15px"
  },

  link: {
    color: "rgb(0, 0, 238)",
    cursor: "pointer",
    textDecoration: "underline",
    "&:active": {
      color: "rgb(255, 0, 0)"
    },
  },
  
})

const EmailActivation = ({ classes, email }) => {
  const resend = () => {}
  return (
    <Paper className={classes.paper}>
      <Typography className={classes.title} variant="h3" align="center">
        Thank you for registration!
      </Typography>
      <Typography className={classes.p}>
        Weâ€™ve sent a verification email to { email }
        Please use the link provided in the email to activate
        and start using your AppHooks account.
      </Typography>
      <Typography className={classes.p}>
        You can <span className={classes.link} onClick={resend}>re-send a verification mail.</span>
      </Typography>
    </Paper>
  ) 
}

EmailActivation.propTypes = {
  email: PropTypes.string.isRequired
}

export default withBackground(withStyles(styles)(EmailActivation))
