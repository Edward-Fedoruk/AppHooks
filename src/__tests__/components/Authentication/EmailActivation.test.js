import React from "react"
import { shallow, mount } from "enzyme"
import sinon from "sinon"
import EmailActivation from "../../../components/Authentication/EmailActivation"

const location = {
  state: {
    userData: { userEmail: "email" },
  },
}

xdescribe('EmailActivation', () => {
  it('should be defined', () => {
    expect(EmailActivation).toBeDefined()
  })

  it("should render correctly", () => {
    const spy = sinon.spy()
    const tree = shallow(
      <EmailActivation.WrappedComponent classes={{}} location={location} reSenEmail={spy} />
    )
    expect(tree).toMatchSnapshot()
  })

  it("calls reSendEmail on click", () => {
    const spy = sinon.spy()

    const tree = shallow(
      <EmailActivation.WrappedComponent classes={{}} location={location} reSendEmail={spy} />
    )

    tree
      .find("span")
      .first()
      .simulate("click")

    expect(spy.calledOnce).toBe(true)
  })
})
