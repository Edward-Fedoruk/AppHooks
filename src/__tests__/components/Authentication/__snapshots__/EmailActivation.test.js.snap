// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailActivation should render correctly 1`] = `
<WithStyles(Paper)>
  <WithStyles(Typography)
    align="center"
    variant="h3"
  >
    Thank you for registration!
  </WithStyles(Typography)>
  <WithStyles(Typography)>
    Weâ€™ve sent a verification email to 
    email
    . Please use the link provided in the email to activate and start using your AppHooks account.
  </WithStyles(Typography)>
  <WithStyles(Typography)>
    You can 
    <span
      onClick={[Function]}
    >
      re-send a verification mail.
    </span>
  </WithStyles(Typography)>
</WithStyles(Paper)>
`;
